flink-ml-core/src/test/java/org/apache/flink/ml/common/window/WindowsTest.java

```diff
@@ -0,0 +1,260 @@
+                DataStreamUtils.windowAllProcess(
+                        inputStreamWithProcessingTimeGap,
+                        ProcessingTimeTumblingWindows.of(Time.milliseconds(100)),
+                        new CreateAllWindowBatchFunction<>());
+        List<List<Long>> actualBatches = IteratorUtils.toList(outputStream.executeAndCollect());
+        assertTrue(actualBatches.size() > 1);
+        List<Long> mergedBatches = new ArrayList<>();
+        for (List<Long> batch : actualBatches) {
+            mergedBatches.addAll(batch);
+        }
+        assertTrue(mergedBatches.containsAll(inputData.subList(0, RECORD_NUM - 1)));
+    }
+
+    @Test
+    public void testEventTimeTumblingWindows() throws Exception {
+        DataStream<List<Long>> outputStream =
+                DataStreamUtils.windowAllProcess(
+                        inputStreamWithEventTime,
+                        EventTimeTumblingWindows.of(Time.milliseconds(RECORD_NUM / 7)),
+                        new CreateAllWindowBatchFunction<>());
+        List<List<Long>> actualBatches = IteratorUtils.toList(outputStream.executeAndCollect());
+        assertEquals(8, actualBatches.size());
+        List<Long> mergedBatches = new ArrayList<>();
+        for (List<Long> batch : actualBatches) {
+            mergedBatches.addAll(batch);
+        }
+        assertEquals(RECORD_NUM, mergedBatches.size());
+        assertEquals(new HashSet<>(inputData), new HashSet<>(mergedBatches));
+    }
+
+    @Test
+    public void testProcessingTimeSessionWindows() throws Exception {
+        DataStream<List<Long>> outputStream =
+                DataStreamUtils.windowAllProcess(
+                        inputStreamWithProcessingTimeGap,
+                        ProcessingTimeSessionWindows.withGap(Time.milliseconds(100)),
+                        new CreateAllWindowBatchFunction<>());
+        List<List<Long>> actualBatches = IteratorUtils.toList(outputStream.executeAndCollect());
+        assertTrue(actualBatches.size() > 1);
+        List<Long> mergedBatches = new ArrayList<>();
+        for (List<Long> batch : actualBatches) {
+            mergedBatches.addAll(batch);
+        }
+        assertTrue(mergedBatches.containsAll(inputData.subList(0, RECORD_NUM - 1)));
+    }
+
+    @Test
+    public void testEventTimeSessionWindows() throws Exception {
+        DataStream<List<Long>> outputStream =
+                DataStreamUtils.windowAllProcess(
+                        inputStreamWithEventTime,
+                        EventTimeSessionWindows.withGap(Time.milliseconds(RECORD_NUM / 7)),
+                        new CreateAllWindowBatchFunction<>());
+        List<List<Long>> actualBatches = IteratorUtils.toList(outputStream.executeAndCollect());
+        assertEquals(1, actualBatches.size());
+        assertEquals(new HashSet<>(inputData), new HashSet<>(actualBatches.get(0)));
+    }
+
+    @Test
+    public void testSaveLoadWindowsParams() throws Exception {
```

Comment:
It might be more readable to move this test into `StageTest.java` for consistency with `test_stage.py`, and keep the practice that all tests regarding the parameters' save/load can be found in StageTest.java.